File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/uncomment/direct/Java/Go/s824230885_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	N := nextInt(scanner)
	list := make([]int, N)
	for i := 0; i < N; i++ {
		list[i] = nextInt(scanner)
	}

	max := 0
	output := 0
	for _, value := range list {
		if max <= value {
			max = value
			output++
		}
	}
	fmt.Println(output)
}

func nextInt(scanner *bufio.Scanner) int {
	scanner.Scan()
	num, _ := strconv.Atoi(scanner.Text())
	return num
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/direct/comment/uncomment/direct/Java/C/s824230885_commented.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_INPUT_SIZE 1024

typedef struct {
    int *array;
    size_t size;
    size_t capacity;
} ArrayList;

void init(ArrayList *list, size_t initial_capacity) {
    list->array = (int *)malloc(initial_capacity * sizeof(int));
    list->size = 0;
    list->capacity = initial_capacity;
}

void append(ArrayList *list, int value) {
    if (list->size == list->capacity) {
        list->capacity *= 2;
        list->array = (int *)realloc(list->array, list->capacity * sizeof(int));
    }
    list->array[list->size++] = value;
}

void free_list(ArrayList *list) {
    free(list->array);
}

int next_int() {
    int num;
    scanf("%d", &num);
    return num;
}

ArrayList next_int_array() {
    ArrayList list;
    init(&list, 16); // Initial capacity of 16
    char c;
    while ((c = getchar()) != '\n') {
        ungetc(c, stdin);
        append(&list, next_int());
    }
    return list;
}

void solve() {
    int N = next_int();
    ArrayList list = next_int_array();
    int max = 0;
    int output = 0;
    for (int i = 0; i < N; i++) {
        if (max <= list.array[i]) {
            max = list.array[i];
            output++;
        }
    }
    printf("%d\n", output);
    free_list(&list);
}

int main() {
    solve();
    return 0;
}

// End of Code
